src/CashuWallet.ts

```diff
@@ -33,18 +35,23 @@ import {
 class CashuWallet {
 	private _keys: MintKeys;
 	private _keysetId = '';
+	private _seed: Uint8Array | undefined;
 	mint: CashuMint;
 
 	/**
 	 * @param keys public keys from the mint
 	 * @param mint Cashu mint instance is used to make api calls
+	 * @param mnemonic mnemonic seed phrase to initial derivation key for this wallets deterministic secrets
 	 */
-	constructor(mint: CashuMint, keys?: MintKeys) {
+	constructor(mint: CashuMint, keys?: MintKeys, mnemonic?: string) {
 		this._keys = keys || {};
 		this.mint = mint;
 		if (keys) {
 			this._keysetId = deriveKeysetId(this._keys);
 		}
+		if (mnemonic) {
+			this._seed = deriveSeedFromMnemonic(mnemonic);
```

Comment:
What happens if `mnemonic` is not a valid bip39 string?

Reply:
nothing! but we should check for that. good catch!